Modules/Filtering/Smoothing/test/itkFFTDiscreteGaussianImageFilterTest.cxx

```diff
@@ -0,0 +1,70 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#include <iostream>
+
+#include "itkFFTDiscreteGaussianImageFilter.h"
+#include "itkImageFileReader.h"
+#include "itkImageFileWriter.h"
+#include "itkTestingMacros.h"
+
+int
+itkFFTDiscreteGaussianImageFilterTest(int argc, char * argv[])
+{
+  if (argc != 5)
+  {
+    std::cerr << "Missing parameters." << std::endl;
+    std::cerr << "Usage:" << std::endl;
+    std::cerr << itkNameOfTestExecutableMacro(argv) << " inputFilename sigma kernelWidth outputFilename filterDimensionality" << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  using ScalarPixelType = float;
+  const size_t ImageDimension = 2;
+  using ImageType = itk::Image<ScalarPixelType, ImageDimension>;
+
+  double       sigma = std::stod(argv[2]);
+  unsigned int kernelWidth = std::stoi(argv[3]);
+  unsigned int filterDimensionality = (argc < 6 ? ImageDimension : std::stoi(argv[5]));
+
+  using ReaderType = itk::ImageFileReader<ImageType>;
+  auto reader = ReaderType::New();
+  reader->SetFileName(argv[1]);
+  reader->Update();
+
+  using FilterType = itk::FFTDiscreteGaussianImageFilter<ImageType, ImageType>;
+  auto filter = FilterType::New();
+  ITK_EXERCISE_BASIC_OBJECT_METHODS(filter, FFTDiscreteGaussianImageFilter, DiscreteGaussianImageFilter);
+
+  filter->SetInput(reader->GetOutput());
+  filter->SetVariance(sigma);
+  filter->SetMaximumKernelWidth(kernelWidth);
+  filter->SetUseImageSpacing(true);
+  filter->Update();
+
+  using WriterType = itk::ImageFileWriter<ImageType>;
+
+  auto writer = WriterType::New();
```

Comment:
Why not use `itk::WriteImage(filter->GetOutput(), argv[4, true);`? We are not trying to demonstrate streaming, or other advanced usage.